src/commands/deployment/execute.rs

```diff
@@ -655,35 +622,167 @@ pub async fn handle_execute(
+                    }
 
-            TwoColumns {
-                left: build_artifact_name,
-                right: commits,
-                left_width: 0,
-            }
-        })
-        .collect();
+                    if *build_artifact == build_artifact_name {
+                        ThreeColumns {
+                            left: build_artifact_name,
+                            middle: "*".to_string(),
+                            right: commits,
+                            left_width: 0,
+                        }
+                    } else {
+                        ThreeColumns {
+                            left: build_artifact_name,
+                            middle: "".to_string(),
+                            right: commits,
+                            left_width: 0,
+                        }
+                    }
+                })
+                .collect();
+            build_selection.extend(jenkins_builds)
+        }
+    };
+
+    build_selection.iter_mut().for_each(|build| {
+        build.left_width = width;
+    });
+
+    build_selection
+}
+
+fn generate_two_columns_build_selection(
+    pipeline_type: &PipelineType,
+    cd_pipeline: &CdPipelineWithBuilds,
+) -> Vec<TwoColumns> {
+    let mut width = 0;
+
+    let mut two_columns: Vec<TwoColumns> = match pipeline_type {
+        PipelineType::Github => cd_pipeline
+            .github_builds
+            .iter()
+            .map(|build| {
+                let commits: Vec<String> = build
+                    .commits
+                    .iter()
+                    .map(|commit| commit.message_headline.clone())
+                    .collect();
+
+                let build_artifact_name = build.build_artifact_name.clone();
+                if build_artifact_name.len() > width {
+                    width = build_artifact_name.len();
+                }
+
+                TwoColumns {
+                    left: build_artifact_name,
+                    right: commits,
+                    left_width: 0,
+                }
+            })
+            .collect(),
+        PipelineType::Jenkins => cd_pipeline
+            .jenkins_builds
+            .iter()
+            .map(|build| {
+                let commits: Vec<String> = build
+                    .commits
+                    .iter()
+                    .map(|commit| commit.message_headline.clone())
+                    .collect();
+
+                let build_artifact_name = build.build_artifact_name.clone();
+                if build_artifact_name.len() > width {
+                    width = build_artifact_name.len();
+                }
+
+                TwoColumns {
+                    left: build_artifact_name,
+                    right: commits,
+                    left_width: 0,
+                }
+            })
+            .collect(),
+    };
 
     two_columns
         .iter_mut()
         .for_each(|each| each.left_width = width);
+
     two_columns
 }
```

Comment:
I think we could just pass the builds to the function, so we don't need to check the pipeline type. (Provided `github_builds` and `jenkins_builds` are the same type.)

Reply:
Thanks, this helped me to look into this in a diff angle. now code looks much cleaner.

Still i kept the filteration within the func to reduce the logic in the main function. Main func has alot of code already.  :')